doc/developer_guide/developer_guide.rst

```diff
@@ -0,0 +1,341 @@
+user specific changes that could prevent the script client from working
+properly.
+
+The following graph shows the default build steps and their
+dependencies.
+
+.. image:: ./images/image-dependencies.png
+  :width: 400
+  :alt: Build-Step dependencies
+
+A dependency between build steps can be either a FROM or COPY
+dependencies. A FROM dependency means that the target of the arrow uses
+the source of the arrow as base image. A COPY dependency means that the
+target of the arrow `copies
+parts <https://docs.docker.com/develop/develop-images/multistage-build/>`__
+of the source of the arrow.
+
+All steps with the string “build_run” in their name, either run the
+build for the script client or at least inherit from an image which had
+built it. As such these images contain all necessary tools to rebuild
+the script client for debugging purposes.
+
+How do we Define Build Steps for a Flavor?
+------------------------------------------
+
+Each flavor has a build_steps.py file in the /flavor_base directory
+which defines the build steps as classes which inherit from
+DockerFlavorAnalyzeImageTask. For example:
+
+.. code:: python
+
+   class AnalyzeBuildRun(DockerFlavorAnalyzeImageTask):
+
+       def get_build_step(self) -> str:
+       # name of the build step, which defines the directory name
+       # for the build context of this image and gets used for the
+       # build boundaries
+           return "build_run"
+
+       def requires_tasks(self):
+       # other build steps the current build step depends on, the keys used here,
+       # get replaced in your dockerfile with the actual image names of your dependencies
+           return {"build_deps": AnalyzeBuildDeps(flavor_path=self.flavor_path),
+                   "language_deps": AnalyzeLanguageDeps(flavor_path=self.flavor_path)}
+
+       def get_additional_build_directories_mapping(self) -> Dict[str, str]:
+       # additional build directories or files which are specific to the build step
+           return {"src": "src"}
+
+       def get_path_in_flavor(self):
+       # to get the path to the build context of the build step within the flavor path
+           return "flavor_base"
+
+       def get_image_changing_build_arguments(self):
+       # optional: build arguments which might change the image content
+           return dict()
+
+       def get_transparent_build_arguments(self):
+       # optional: build arguments which won't change the image content
+           return dict()
+
+How Does Caching Work?
+----------------------
+
+Exaslct was built with caching in mind, because building a flavor might
+take very long, and many build steps don’t change that often.
+Furthermore, an end user most likely only changes the build-step
+flavor-customization which is designed to have a minimal impact on all
+other build steps.
+
+Which Caches are Available?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+EXASLCT provides three types of caching: \* docker images managed by the
```

Comment:
not sure if the \* are rendering as you'd like. They look a bit odd to me.
:warning: this is in a few places, but only marked here.

![image](https://github.com/user-attachments/assets/685a01b3-909d-4f54-bc98-02e18d339b85)

Reply:
You are right. I used pandoc to convert the user guide and developer guide from markdown to rst. Appearantly it converted the bullet points incorrectly.

Fixed it.